packages/lab/src/overlay/Overlay.tsx

```diff
@@ -1,100 +1,154 @@
 
 export const Overlay = forwardRef<HTMLDivElement, OverlayProps>(
-  (
-    {
-      adaExceptions: { showClose } = {
-        showClose: true,
-      },
-      arrowProps,
+  function Overlay(props, ref) {
+    const {
       children,
       className,
-      open,
-      onOpenChange,
-      variant = "primary",
+      open: openProp,
+      onOpenChange: onOpenChangeProp,
+      content,
+      placement = "top",
+      onClose,
       ...rest
-    },
-    ref
-  ) => {
+    } = props;
+
     const targetWindow = useWindow();
     useComponentCssInjection({
       testId: "salt-overlay",
       css: overlayCSS,
       window: targetWindow,
     });
 
-    // Will need to figure out a better way to assign popper id's for the electron windows
-    const id = useId();
-    const Window = usePortalWindow();
+    const {
+      arrowProps,
+      open,
+      onOpenChange,
+      context,
+      floating,
+      reference,
+      getTriggerProps,
+      getOverlayProps,
+      floatingStyles,
+    } = useOverlay({
+      open: openProp,
+      placement,
+      onOpenChange: onOpenChangeProp,
+    });
 
-    const handleCloseButton = () => {
-      onOpenChange?.(false);
-    };
+    const { Component: FloatingComponent } = useFloatingComponent();
 
-    if (!open) {
-      return null;
-    }
+    const triggerRef = useForkRef(
+      // @ts-ignore error TS2339
+      isValidElement(children) ? children.ref : null,
+      reference
+    );
+
+    const floatingRef = useForkRef<HTMLDivElement>(floating, ref);
+
+    const handleCloseButton = (event: SyntheticEvent) => {
+      onOpenChange(false);
+      onClose?.(event);
+    };
 
     return (
-      <Portal>
-        <Window
-          className={clsx(withBaseName(), className, {
-            [withBaseName(variant)]: variant === "secondary",
+      <>
+        {isValidElement(children) &&
+          cloneElement(children, {
+            ...mergeProps(getTriggerProps(), children.props),
+            ref: triggerRef,
           })}
-          id={id}
-          ref={ref}
-          {...rest}
-        >
-          <div
-            className={clsx(withBaseName("content"))}
-            data-testid="overlay-content"
-          >
-            {showClose && (
-              <Button
-                onClick={handleCloseButton}
-                className={withBaseName("close")}
-                variant="secondary"
+
+        {open && (
+          <FloatingOverlay>
+            <FloatingComponent
+              ref={floatingRef}
+              open={open}
+              className={withBaseName()}
+              {...getOverlayProps()}
+              {...floatingStyles()}
+              // @ts-ignore missing 'children' property
+              focusManagerProps={{
+                context: context,
+              }}
+            >
+              <div
+                className={clsx(withBaseName("container"), className)}
+                aria-modal="true"
```

Comment:
Just to add it here too, aria-modal  should be where the role=dialog is and so should aria-labelledby and aria-describedby

Reply:
updated